src/components/course/CoursePage.jsx

```diff
@@ -1,12 +1,35 @@
-import React from 'react';
+import React, { useContext, useEffect } from 'react';
+import Cookies from 'universal-cookie';
+import qs from 'query-string';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { getConfig } from '@edx/frontend-platform/config';
+import { AppContext } from '@edx/frontend-platform/react';
 
 import Course from './Course';
 import AuthenticatedUserSubsidyPage from '../app/AuthenticatedUserSubsidyPage';
+import { loginRefresh } from '../../utils/common';
 
 export default function CoursePage() {
-  return (
-    <AuthenticatedUserSubsidyPage>
-      <Course />
-    </AuthenticatedUserSubsidyPage>
-  );
+  const { authenticatedUser } = useContext(AppContext);
+  const { roles } = authenticatedUser;
+
+  // If the user has not refreshed their JWT since they created their account,
+  // we should refresh it so that they'll have appropriate roles (if available),
+  // and thus, have any appropriate permissions when making downstream requests.
+  if (!roles || Object.keys(roles).length == 0) {
```

Comment:
`roles` is an array. If there are no `roles`, it will be an empty array. The following conditional should suffice:

```js
if (roles.length === 0) { ... }
```